test/unit/EventEmitter.test.js

```diff
@@ -60,46 +60,11 @@ describe('EventEmitter', () => {
       ev.emit('sample')
       expect(cnt).toBe(1)
     })
-
-    it('should throw when try to add the same listener multiple times', () => {
-      const ev = new EventEmitter()
-      const cb = () => {}
-
-      ev.on('sample', cb)
-
-      const run = () => ev.on('sample', cb)
-
-      expect(run).toThrow(/Listener already exists for router event: `sample`/)
-    })
-
-    it('should support chaining like the nodejs EventEmitter', () => {
```

Comment:
Isn't this a breaking change? It seems like doing a `return this;` in `mitt` would be a small change that would avoid surprising those used to node's `EventEmitter`.

Reply:
I'm afraid it would make `mitt` much bigger in size, because `mitt` is not a class but a function.

Reply:
That doesn't matter; `mitt()` returns an _object_, so `this` would refer to that object, and this would allow for the same type of method chaining.

Reply:
You're right, I forgot about that 😅

@timneutkens Should we return `this` in `on` and `off`,  ?

It adds ~25 bytes in the minified code (not gzipped).

Reply:
Mitt doesn't currently return `this` because the context of the method isn't guaranteed:

```js
const { emit, on, off } = mitt()

on('foo', console.log)
    .emit('foo')
```

The value of `this` when `on` is invoked is window/undefined.

This is also true when mixing mitt into an object:

```js
router = Object.assign(Router(), mitt())
router.on() // this is Router instance
```

If chaining is important as part of the public API here, I think it's best to implement the return semantics here.